plugin/src/com/microsoft/alm/plugin/external/visualstudio/VisualStudioTfvcClient.java

```diff
@@ -0,0 +1,110 @@
+
+    @Nullable
+    public static Path getOrDetectPath(PropertyService propertyService) {
+        String path = propertyService.getProperty(PropertyService.PROP_VISUAL_STUDIO_TF_CLIENT_PATH);
+        if (!StringUtil.isEmpty(path)) {
+            return Paths.get(path);
+        }
+
+        return detectClientPath();
+    }
+
+    /**
+     * Tries to detect path to the latest available TFVC client from a set of hardcoded locations.
+     *
+     * @return detected path or null.
+     */
+    @Nullable
+    public static Path detectClientPath() {
+        if (!SystemInfo.isWindows) return null;
+        String programFilesPath = SystemInfo.is32Bit ? System.getenv("ProgramFiles") : System.getenv("ProgramFiles(x86)");
+        if (StringUtils.isEmpty(programFilesPath))
+            programFilesPath = "C:\\Program Files (x86)";
+        Path visualStudioPath = Paths.get(programFilesPath, "Microsoft Visual Studio");
+        for (String vsVersion : SUPPORTED_VS_VERSIONS) {
+            Path versionPath = visualStudioPath.resolve(vsVersion);
+            if (!versionPath.toFile().isDirectory()) continue;
+            for (String vsEdition : SUPPORTED_VS_EDITIONS) {
+                Path editionPath = versionPath.resolve(vsEdition);
+                Path tfExePath = editionPath.resolve(EDITION_RELATIVE_TF_EXE_PATH);
+                if (tfExePath.toFile().isFile()) return tfExePath;
+            }
+        }
+
+        return null;
+    }
+
+    @NotNull
+    public static CompletionStage<Void> checkVersionAsync(Path visualStudioClientPath) {
+        return VisualStudioTfvcCommands.getVersionAsync(visualStudioClientPath).thenAccept(version -> {
+            if (version == null)
+                throw new VisualStudioClientVersionException(ToolVersion.UNKNOWN, MINIMAL_SUPPORTED_VERSION);
+
+            if (MINIMAL_SUPPORTED_VERSION.compare(version) > 0)
+                throw new VisualStudioClientVersionException(version, MINIMAL_SUPPORTED_VERSION);
+        });
+    }
+
+    static List<String> executeClientAndGetOutput(
+            @NotNull Path clientPath,
+            @Nullable Path workingDirectory,
+            @NotNull List<String> arguments) throws IOException, InterruptedException {
+        ourLogger.info("Executing VS client: " + clientPath + ", args: " + StringUtils.join(arguments, ','));
+        List<String> command = new ArrayList<>(arguments.size() + 1);
+        command.add(clientPath.toString());
+        command.addAll(arguments);
+
+        Process client = new ProcessBuilder()
+                .command(command)
+                .directory(workingDirectory == null ? null : workingDirectory.toFile())
+                .start();
+
+        List<String> output = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(client.getInputStream()))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                ourLogger.info("VS client stdout: " + line);
```

Comment:
Are you sure about the "info" logging level?

Reply:
Yes, I think this is a good idea. The VS client isn't too verbose, and won't be called too often (only on manually triggered import sequence). 5-10 lines of output wouldn't hurt.

Also, we generally log all external tool stdout to `info`.